h2o-core/src/main/java/hex/Model.java

```diff
@@ -1583,6 +1596,12 @@ public Frame score(Frame fr, String destination_key, Job j, boolean computeMetri
   }
   public Frame score(Frame fr, String destination_key, Job j, boolean computeMetrics, CFuncRef customMetricFunc) throws IllegalArgumentException {
     Frame adaptFr = new Frame(fr);
+    List<Frame> tmpFrames = new ArrayList<>();
+    tmpFrames.add(adaptFr);
+    applyPreprocessors(adaptFr, tmpFrames);
+    //PUBDEV-7775: we need to apply encoding independently from adaptTestForTrain, otherwise previously encoded columns are removed during adaptation
+    //if enabling this, call to adaptTesttoTrain below should use catEncoded=true, however it breaks several tests currently: necessary only for TE in combination with encoders like OHE
+//    encodeCategoricals(adaptFr, tmpFrames); 
```

Comment:
commented out code
formatting
generally comments like this have little value in the long term, if I understand this correctly `applyPreprocessors(adaptFr, tmpFrames);` now does `encodeCategoricals`, if thats the case I would be fine with
`applyPreprocessors(adaptFr, tmpFrames);  // also does categorical encoding`

Reply:
I'll remove this, I have a separate branch now where I started to experiment with this PUBDEV-7775 issue.
Some encodings like binary and ohe are actually broken in some conditions (especially the Mojo), and I was trying to get them working with TE on top, which is just too challening for this PR, and I'm 99% sure that `adaptTesttoTrain` with its crazy cyclomatic complexity and without any clear separation of concerns is responsible for this and needs to be rethought...